java/algorithm/src/main/java/org/apache/arrow/algorithm/sort/FixedWidthOutOfPlaceVectorSorter.java

```diff
@@ -44,6 +45,13 @@ public void sortOutOfPlace(V srcVector, V dstVector, VectorValueComparator<V> co
     ArrowBuf dstValidityBuffer = dstVector.getValidityBuffer();
     ArrowBuf dstValueBuffer = dstVector.getDataBuffer();
 
+    // check buffer size
+    Preconditions.checkArgument(dstValidityBuffer.capacity() * 8 >= srcVector.getValueCount(),
+        "No enough capacity for the validity buffer of the dst vector");
```

Comment:
nit: here and other locations "No" -> "Not".  It might also pay to include actual values using format strings in the error message.

Reply:
Good point. Thank you.
I have revised the code accordingly.